### DISCUSSION TITLE
Understanding Higher Latency in Go Compared to Python

### DISCUSSION THREAD

**Discussion by gourangabani** [gorules/zen-go#39] [Q&A]
Hi everyone,

I've noticed that when using the Zen Engine, the latency appears to be higher in Go than in Python. I'm trying to understand the reasons behind this performance discrepancy.

Has anyone else encountered this issue? What factors could be contributing to the higher latency in Go? Is it related to Go's concurrency model, garbage collection, or perhaps something specific to the CGO libraries used by the Zen Engine implementation in Go?

Any insights, benchmarks, or experiences would be greatly appreciated!

Thanks!

**Comment by ivanmiletic** [comment]
Hi @gourangabani are you able to share more information, when you say latency what do you mean exactly?

Is it engine performance or something else? We have some good benchmarks in Go (monitoring p99s on high load) for internal purposes.

**Reply by gourangabani** [reply]
Hi @ivanmiletic , 

I've noticed an interesting behavior with my Zen-Go application. When running on a single CPU, the Zen Engine in Python outperformed Zen-Go. However, after increasing the CPU count from 1 to 2 for the Zen-Go application, I observed a roughly 60% reduction in latency. Oddly, the CPU utilization wasn't high when the application was running on a single CPU.

Do you have any insights into why this might be happening? Are there specific factors related to Go's concurrency model or resource management that could explain this improvement with additional CPUs?

Thanks for your help!

**Reply by ivanmiletic** [reply]
Hi @gourangabani we will investigate this after we release new big features this week.

**Reply by gourangabani** [reply]
Sure, thank you!
